Nokia 7210

hooligan-sa opened the pull request:
Creates device model files for:
7210-SAS-M-24F-2XFP
7210-SAS-M-1
7210-SAS-Mxp-22F2C-4SFPP
7210-SAS-T-12F10T-4XFP

danner26 commented on device-types/Nokia/7210-SAS-M-1.yaml:
```diff
@@ -0,0 +1,67 @@
+part_number: 3HE04699AAAB01
+u_height: 2
+is_full_depth: false
+subdevice_role: parent
```
`subdevice_role: parent` is only to be used if this device is either a parent (i.e. a chassis device that houses other devices), or or a child (i.e. this device slots into a chassis device). Generally parent devices have `device-bays` defined for their children devices.

**If this is not actually a parent device, please remove this definition**

hooligan-sa replied:
Added device-bays

danner26 replied:
Which device does this house?

danner26 commented on device-types/Nokia/7210-SAS-M-24F-2XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+part_number: 3HE05828AAAB01
+u_height: 2
+is_full_depth: false
+subdevice_role: parent
```
`subdevice_role: parent` is only to be used if this device is either a parent (i.e. a chassis device that houses other devices), or or a child (i.e. this device slots into a chassis device). Generally parent devices have `device-bays` defined for their children devices.

**If this is not actually a parent device, please remove this definition**

danner26 commented on device-types/Nokia/7210-SAS-Mxp-22F2C-4SFPP.yaml:
```diff
@@ -0,0 +1,75 @@
+part_number: 3HE10075ABRC01
+u_height: 2
+is_full_depth: false
+subdevice_role: parent
```
`subdevice_role: parent` is only to be used if this device is either a parent (i.e. a chassis device that houses other devices), or or a child (i.e. this device slots into a chassis device). Generally parent devices have `device-bays` defined for their children devices.

**If this is not actually a parent device, please remove this definition**

hooligan-sa replied:
Removed

danner26 commented on device-types/Nokia/7210-SAS-T-12F10T-4XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+part_number: 3HE08116AAAB01
+u_height: 2
+is_full_depth: false
+subdevice_role: parent
```
`subdevice_role: parent` is only to be used if this device is either a parent (i.e. a chassis device that houses other devices), or or a child (i.e. this device slots into a chassis device). Generally parent devices have `device-bays` defined for their children devices.

**If this is not actually a parent device, please remove this definition**

danner26 commented on device-types/Nokia/7210-SAS-M-24F-2XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+  - name: Console
+    type: rj-45
+power-ports:
+  - name: PS1
+    type: iec-60320-c14
+    maximum_draw: 450
+  - name: PS2
+    type: iec-60320-c14
+    maximum_draw: 450
```
Is the power on these modular? Or are the power ports not removable?

hooligan-sa replied:
Chose to remove the "power-ports" section and add the module-bays only

danner26 commented on device-types/Nokia/7210-SAS-T-12F10T-4XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+  - name: Console
+    type: rj-45
+power-ports:
+  - name: PS1
+    type: iec-60320-c14
+    maximum_draw: 450
+  - name: PS2
+    type: iec-60320-c14
+    maximum_draw: 450
```
Is the power on these modular? Or are the power ports not removable?

danner26 commented on device-types/Nokia/7210-SAS-M-1.yaml:
```diff
@@ -0,0 +1,67 @@
+slug: nokia-7210-sas-m-1
+part_number: 3HE04699AAAB01
+u_height: 2
+is_full_depth: false
```
Please add the `airflow` attribute if you have it available

hooligan-sa replied:
Added

danner26 commented on device-types/Nokia/7210-SAS-M-24F-2XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+slug: nokia-7210-sas-m-24f-2xfp
+part_number: 3HE05828AAAB01
+u_height: 2
+is_full_depth: false
```
Please add the `airflow` attribute if you have it available

hooligan-sa replied:
Added

danner26 commented on device-types/Nokia/7210-SAS-Mxp-22F2C-4SFPP.yaml:
```diff
@@ -0,0 +1,75 @@
+slug: nokia-7210-sas-mxp-22f2c-4sfpp
+part_number: 3HE10075ABRC01
+u_height: 2
+is_full_depth: false
```
Please add the `airflow` attribute if you have it available

danner26 commented on device-types/Nokia/7210-SAS-T-12F10T-4XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+slug: nokia-7210-sas-t-12f10t-4xfp
+part_number: 3HE08116AAAB01
+u_height: 2
+is_full_depth: false
```
Please add the `airflow` attribute if you have it available

danner26 commented on device-types/Nokia/7210-SAS-T-12F10T-4XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+slug: nokia-7210-sas-t-12f10t-4xfp
+part_number: 3HE08116AAAB01
+u_height: 2
+is_full_depth: false
```
Please add the `weight` and `weight_unit` attribute if you have it available

hooligan-sa replied:
Added

danner26 commented on device-types/Nokia/7210-SAS-M-24F-2XFP.yaml:
```diff
@@ -0,0 +1,71 @@
+slug: nokia-7210-sas-m-24f-2xfp
+part_number: 3HE05828AAAB01
+u_height: 2
+is_full_depth: false
```
Please add the `weight` and `weight_unit` attribute if you have it available

danner26 commented on device-types/Nokia/7210-SAS-M-1.yaml:
```diff
@@ -0,0 +1,67 @@
+slug: nokia-7210-sas-m-1
+part_number: 3HE04699AAAB01
+u_height: 2
+is_full_depth: false
```
Please add the `weight` and `weight_unit` attribute if you have it available

hooligan-sa replied:
Add

hooligan-sa commented on device-types/Nokia/7210-SAS-M-1.yaml:
```diff
@@ -0,0 +1,67 @@
+  - name: Console
+    type: rj-45
+power-ports:
+  - name: PS1
+    type: iec-60320-c14
+    maximum_draw: 450
+  - name: PS2
+    type: iec-60320-c14
+    maximum_draw: 450
```
It's modular, but I haven't done this before. 
Do I remove the "power-ports" section and add module-bay info like below:
module-bays:
  - name: Power Supply 1
  - name: Power Supply 2
  - name: Fan Tray 1

Or do I keep both the power-ports and module-bay sections?

hooligan-sa replied:
Chose to remove the "power-ports" section and add the module-bays only

hooligan-sa commented on device-types/Nokia/7210-SAS-Mxp-22F2C-4SFPP.yaml:
```diff
@@ -0,0 +1,75 @@
+slug: nokia-7210-sas-mxp-22f2c-4sfpp
+part_number: 3HE10075ABRC01
+u_height: 2
+is_full_depth: false
```
Added

hooligan-sa commented on device-types/Nokia/7210-SAS-Mxp-22F2C-4SFPP.yaml:
```diff
@@ -0,0 +1,75 @@
+    maximum_draw: 450
+  - name: PS2
+    type: iec-60320-c14
+    maximum_draw: 450
```
Chose to remove the "power-ports" section and add the module-bays only

danner26 commented:
Which devices do these act as a chassis for? What is `mda2`?

hooligan-sa commented:
> Which devices do these act as a chassis for? What is `mda2`?

Mea culpa, I didn't know the difference between 'device-bays' and 'module-bays' especially related to a 'subdevice_role'. 
I've corrected this and added 'mda2' as a module bay - That is the internal card type slot for the additional module that I'm aware of

danner26 commented on device-types/Nokia/7210-SAS-M-1.yaml:
```diff
@@ -0,0 +1,71 @@
+  - name: 1/2/2
+    type: 10gbase-x-xfp
+module-bays:
+  - name: mda2
+  - name: Power Supply 1
+  - name: Power Supply 2
+  - name: Fan Tray 1
```
Generally you should also be adding `position` attributes to module-bays. When a module is imported in NetBox, you will need to use the position value

This is done automatically when importing a module that has its interface/port name that contains `{module}`.

danner26 commented on device-types/Nokia/7210-SAS-M-24F-2XFP.yaml:
```diff
@@ -0,0 +1,75 @@
+  - name: 1/2/2
+    type: 10gbase-x-xfp
+module-bays:
+  - name: mda2
+  - name: Power Supply 1
+  - name: Power Supply 2
+  - name: Fan Tray 1
```
Generally you should also be adding `position` attributes to module-bays. When a module is imported in NetBox, you will need to use the position value

This is done automatically when importing a module that has its interface/port name that contains `{module}`.

danner26 commented on device-types/Nokia/7210-SAS-Mxp-22F2C-4SFPP.yaml:
```diff
@@ -0,0 +1,74 @@
+  - name: 1/1/28
+    type: 10gbase-x-sfpp
+module-bays:
+  - name: Power Supply 1
+  - name: Power Supply 2
+  - name: Fan Tray 1
```
Generally you should also be adding `position` attributes to module-bays. When a module is imported in NetBox, you will need to use the position value

This is done automatically when importing a module that has its interface/port name that contains `{module}`.

danner26 commented on device-types/Nokia/7210-SAS-T-12F10T-4XFP.yaml:
```diff
@@ -0,0 +1,70 @@
+  - name: 1/1/26
+    type: 10gbase-x-xfp
+module-bays:
+  - name: Power Supply 1
+  - name: Power Supply 2
+  - name: Fan Tray 1
```
Generally you should also be adding `position` attributes to module-bays. When a module is imported in NetBox, you will need to use the position value

This is done automatically when importing a module that has its interface/port name that contains `{module}`.

danner26 reviewed (changes requested):
Please review my other comments. Also, is the `mda2` module already in the module-types folder? If not, please create it and add that to this PR as well. We need to make sure when adding new hardware/module-bays that the assicated modules exist for importing as well

danner26 commented:
@hooligan-sa generally for modules we use the {module} definition in the name, so that the position value is used when placing the module

This is especially helpful when a device has 2 module bays that can have this module. It prevents interface name clashing in NetBox

danner26 commented:
Hello, we have released a new set of tests and features to assist with properly creating definition files for the DeviceType-Library. These tests will ensure that all files conform to a standardized format and contain valid data. Over the coming weeks we will be adding more tests to handle other currently undefined cases. If you notice anything that you believe is a bug, please @ me in a comment and details what you think is incorrect.

Along with additional test cases we have added a feature that will annotate why tests fail. To access these annotations after a test has completed, please navigate to the _Files changed_ tab within your PR. At the very bottom of the page you will see a list of errors along with the filename in question and the error which was found.

With all this being said, if your PR is validated then great! Please @ me me so I can review it further! If your PR is not valid, please review the issues and correct them; in the meantime I will mark your PR as a draft. After correcting the issues and recieving a valid status, please mark the PR as ready to be reviewed. Thank you!
